### PR TITLE
[NS-2]: feat/POST endpoint to finish upload

### PR DESCRIPTION
**Pull request by fabioespinosa**
~DRAFT~

Tested with postman ☑️

### CODE REVIEW

File: lib/server/routes/buckets.js [hunk 1]
```diff
@@ -1396,6 +1398,129 @@ BucketsRouter.prototype.getStorageUsage = function (req, res) {
...
+              return cb(err);
+            }
+
+            createdShardIDs.push(savedShard._id);
+
+            async.eachLimit(
+              contracts,
+              1,
+              (contract, cb) => {
+                const mirrorToCreate = new Mirror({
```
Comment (fabioespinosa) [inline lib/server/routes/buckets.js:1483]: Unsure here if its ok to use `new ` or better to create the method inside storage models

File: lib/server/routes/buckets.js [hunk 2]
```diff
@@ -1396,6 +1398,129 @@ BucketsRouter.prototype.getStorageUsage = function (req, res) {
...
+    return next(new errors.BadRequestError('Missing shards'));
+  }
+
+  if (!Array.isArray(shards)) {
+    return next(new errors.BadRequestError('Shards is not an array'));
+  }
+
+  shards.forEach(({ uuid, hash }) => {
+    if (!validateUUID(uuid)) {
+      return next(new errors.BadRequestError('Invalid UUID'));
```
Comment (sg-gs) [inline lib/server/routes/buckets.js:1422]: Be careful, this is a callback, a return in a callback will not stop the execution of the next iterations nor the `finishUpload` function

File: lib/server/routes/buckets.js [hunk 3]
```diff
@@ -1396,6 +1398,129 @@ BucketsRouter.prototype.getStorageUsage = function (req, res) {
...
+    }
+    if (!hash) {
+      return next(new errors.BadRequestError('Missing hash'));
```
Comment (sg-gs) [inline lib/server/routes/buckets.js:1425]: same here

File: lib/server/routes/buckets.js (hunk 1 above)
Comment (sg-gs) [reply lib/server/routes/buckets.js:1483]: The chosen option with `new` is fine

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: lib/server/routes/buckets.js (hunk 2 above)
Comment (fabioespinosa) [reply lib/server/routes/buckets.js:1422]: fixed

**Comment by fabioespinosa** [comment]
Closing, contained in https://github.com/internxt/bridge/pull/54

**Pull request closed without merging**